pkg/trace/api/dogstatsd_test.go

```diff
@@ -85,15 +87,39 @@ func TestDogStatsDReverseProxyEndToEndUDP(t *testing.T) {
 	cfg.StatsdHost = "127.0.0.1"
 	cfg.StatsdPort = port
 
+	address := fmt.Sprintf("%s:%d", cfg.StatsdHost, cfg.StatsdPort)
+	ln, err := net.ListenPacket("udp", address)
+	if err != nil {
+		t.Fatalf("failed to create listener on %q: %v", address, err)
+	}
+	defer ln.Close()
+
 	receiver := newTestReceiverFromConfig(cfg)
 	proxy := receiver.dogstatsdProxyHandler()
 	require.NotNil(t, proxy)
 	rec := httptest.NewRecorder()
 
-	// Test metrics
-	body := ioutil.NopCloser(bytes.NewBufferString("daemon:666|g|#sometag1:somevalue1,sometag2:somevalue2"))
+	// Send two payloads separated by a newline.
+	payloads := [][]byte{[]byte("daemon:666|g|#sometag1:somevalue1,sometag2:somevalue2"), []byte("_sc|agent.up|0|m:this is fine")}
+	sep := []byte("\n")
+	msg := bytes.Join(payloads, sep)
+	body := ioutil.NopCloser(bytes.NewBuffer(msg))
 	proxy.ServeHTTP(rec, httptest.NewRequest("POST", "/", body))
 	require.Equal(t, http.StatusOK, rec.Code)
+
+	// Check that both payloads were sent over (without a newline).
+	ln.SetReadDeadline(time.Now().Add(2 * time.Second))
```

Comment:
Does this read deadline introduce a possibility for flakiness?

Reply:
It could. When I run it locally it's extremely quick and shouldn't need nearly 2 seconds to run, but I bumped it to 30 seconds just to be sure. Fortunately this doesn't introduce a time.Sleep or anything like that by increasing the deadline, so it's fine to bump it to something we deem reasonable.